### PR TITLE
Move CharToKeyEvents (and friends) into InteractivityBase

### LINKED ISSUE #9105: Types is not bottom-clean; has a dependency on the entire interactivity stack!
Types depends on the VTRedir versions of some keyboard functions.

### PR DESCRIPTION
**Pull request by DHowett**
These functions have a dependency on the "VT Redirected" versions of
VkKeyScanW, MapVirtualKeyW and GetKeyState. Those implementations depend
on the service locator and therefore the entire interactivity stack.

This meant that anybody depending on just Types had to pull in **the
entire host** worth of dependencies (!).

Since these functions are only used in places where we have or are
testing interactivity, it makes sense to consolidate them here.

Closes #9105.

### CODE REVIEW

**Comment by DHowett** [comment]
This unblocks inbox.

File: src/interactivity/base/convert.cpp [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+    if (keyState == invalidKey)
+    {
+        // Determine DBCS character because these character does not know by VkKeyScan.
+        // GetStringTypeW(CT_CTYPE3) & C3_ALPHA can determine all linguistic characters. However, this is
+        // not include symbolic character for DBCS.
+        WORD CharType = 0;
+        GetStringTypeW(CT_CTYPE3, &wch, 1, &CharType);
+
+        if (WI_IsFlagSet(CharType, C3_ALPHA) || GetQuickCharWidth(wch) == CodepointWidth::Wide)
```
Comment (DHowett) [inline src/interactivity/base/convert.cpp:33]: this is highly suspect -- the width of a character should not impact its encoding in key events. HOWEVER: this is NOT NEW CODE and I WILL NOT FIX IT HERE.

**Comment by DHowett** [comment]
Pushing a revision that renames the new file from convert to EventSynthesis

File: src/interactivity/base/convert.cpp (hunk 1 above)
Comment (miniksa) [reply src/interactivity/base/convert.cpp]: Yeah, don't fix it. But you're right. The width... shouldn't... ughhhhhh.

**Review by miniksa — APPROVED** [review]

**Review by PankajBhojwani — APPROVED** [review]

**Pull request merged**